### ISSUE TITLE
BatchNorm with RecurrentInputType

### ISSUE THREAD

**Issue by moinnadeem** [deeplearning4j/deeplearning4j#6697]
I'm getting an error with DL4J 1.0.0 beta 3 when I try and do Keras imports:

Error:
```
Exception in thread "main" java.lang.IllegalStateException: Invalid input type: Batch norm layer expected input of type CNN, CNN Flat or FF, got InputTypeRecurrent(100,timeSeriesLength=2307) for layer index -1, layer name = batch_normalization_4
	at org.deeplearning4j.nn.conf.layers.BatchNormalization.getOutputType(BatchNormalization.java:123)
	at org.deeplearning4j.nn.modelimport.keras.layers.normalization.KerasBatchNormalization.getOutputType(KerasBatchNormalization.java:164)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.inferOutputTypes(KerasModel.java:304)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:147)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:58)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildSequential(KerasModelBuilder.java:322)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:195)
	at com.Databases.DeepBloom.DeepNN.SimpleRNN.main(SimpleRNN.java:25)
```

Model architecture: 
```
model = Sequential()
model.add(GRU(50, input_shape=(sequence_length, num_unique_symbols), activation="tanh"))
model.add(BatchNorm())
model.add(Dropout(0.2))
model.add(Dense(25, activation="tanh"))
model.add(Dropout(0.2))
model.add(Dense(1, activation="sigmoid"))
```

**Comment by ghost** [comment]
I was able to reproduce this with `Keras 2.3.1` and `DL4J 1.0.0 beta 6`.
Error:
```
Exception in thread "main" java.lang.IllegalStateException: Invalid input type: Batch norm layer expected input of type CNN, CNN Flat or FF, got InputTypeRecurrent(256,timeSeriesLength=1493) for layer index -1, layer name = batch_normalization_1
	at org.deeplearning4j.nn.conf.layers.BatchNormalization.getOutputType(BatchNormalization.java:128)
	at org.deeplearning4j.nn.modelimport.keras.layers.normalization.KerasBatchNormalization.getOutputType(KerasBatchNormalization.java:164)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.inferOutputTypes(KerasModel.java:304)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:146)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:57)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildSequential(KerasModelBuilder.java:322)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:223)
```
Model architecture:
```
model = Sequential() 
model.add(Embedding(max_features, embed_size, input_length = maxlen)) 
model.add(Conv1D(256, 8, activation='relu'))
model.add(BatchNormalization())
model.add(MaxPooling1D(3)) 
model.add(Conv1D(128, 4, activation='relu'))
model.add(BatchNormalization())
model.add(GlobalMaxPooling1D())
model.add(Dense(10, activation='softmax'))
model.compile(optimizer=RMSprop(lr=0.01), loss='categorical_crossentropy', metrics=['accuracy'])
```

**Comment by TomatePilatus** [comment]
I encountered the same exception too. I am developing a _convolutional neural network_ acting on **1D data**. So I have 1080 numbers times two channels for an input shape of `[minibatch, 1080, 2]`. I successfully trained the model through Keras on a Jupyter notebook, and I am trying to load it from a `.h5` file by calling `KerasModelImport.importKerasModelAndWeights("path/to/file.h5")`. The first layer is a BatchNormalization and I can not load the model in any way, since the method throws an IllegalStateException with the message: 

```
Invalid input type: Batch norm layer expected input of type CNN, CNN Flat or FF, got InputTypeRecurrent(2,timeSeriesLength=1080,format=NWC) for layer index -1, layer name = batch_normalization_1.
```

I followed the call stack and found that in `KerasInput#getOutputType(InputType...)`, an input of shape size equal to 2 is mapped directly to an `InputType.InputTypeRecurrent` layer. To attain a convolutional input layer—such as `InputType.InputTypeConvolutional` or `InputType.InputTypeConvolutional3D`—one needs at least an input of shape size equal to 3 or 4.

So far, the framework lacks the correct input layer for 1D convolutional networks. Unfortunately, I am not skilled enough to suggests some code changes, but I am looking forward to a new version that may cover this case as well. I really thank the team for their great job.
